README.md

```diff
@@ -53,6 +53,64 @@ You'll need a [browser Buffer polyfill](https://www.npmjs.com/package/browserify
 
 If you want to support node & browsers with the same code, you can use the latter `await` form here everywhere (since awaiting the fixed value in node just returns the value as-is).
 
+
+## Vite and Rollup
+
+For usage in Vite, you have to use the `vite-plugin-wasm` and `static-files` plugins. The config looks like this:
+
+```javascript
+import { defineConfig } from "vite";
+import wasm from "vite-plugin-wasm";
+import { wasm as rollupWasm } from "@rollup/plugin-wasm";
+import { viteStaticCopy } from "vite-plugin-static-copy";
+
+// https://vitejs.dev/config/
+export default defineConfig({
+    plugins: [
+        viteStaticCopy({
+            targets: [
+                {
+                    src: require.resolve("brotli-wasm/pkg.web/brotli_wasm_bg.wasm"),
+                    dest: "."
+                }
+            ]
+        }),
+        wasm(),
+        topLevelAwait()
+    ],
+    build: {
+        minify: false,
+        target: ["esnext"],
+        sourcemap: true,
+        rollupOptions: {
+            treeshake: false,
+            plugins: [rollupWasm()]
+        },
+        ssr: false
+    }
+});
+```
+
+and you can call it in your code like this:
+
+```typescript
+import init, { decompress } from "brotli-wasm/pkg.web/brotli_wasm";
```

Comment:
It would be very nice to have it be `await` able, but `init` can take in an argument to the path of the `wasm` file (which is what makes the vite version work in the first place). 


```javascript
// The web version needs to be explicitly initiated, but we want the API to be the same
// as the bundler version, so we write this small wrapper to make it work.
import init, * as brotliWasm from "./pkg.web/brotli_wasm";

module.exports = initOpts => init(initOpts).then(() => brotliWasm);
```

One option would be a breaking change to the `browser` and `node` versions, and require an explicit call to instantiate the promise.


```javascript
// This makes importing wasm-brotli asynchronous (because of dynamic import).
// This is needed here for Webpack v4 or v5 syncWebAssembly, which don't
// allow synchronous import of WebAssembly from an entrypoint.
module.exports = () => import("./pkg.bundler/brotli_wasm.js");

// We don't want to do this for _all_ usage, because dynamic import isn't
// supported in older node versions.
```

Another option would be to add a named export to all three that is the "callable" version of the promise, and where there is no default export (or it throws) for the `pkg.web` version.